### ISSUE TITLE
Leaks API key on exception

### ISSUE THREAD

**Issue by ricardobranco777** [python-bugzilla/python-bugzilla#187]
The quick way to test it is with no Internet connection:

```
import os
import logging
from bugzilla import Bugzilla
from bugzilla.exceptions import BugzillaError
from requests.exceptions import RequestException

URL = os.getenv("URL", "https://bugzilla.suse.com")
USER = os.getenv("USER", "test")
API_KEY = "MYAPIKEY"

try:
    client = Bugzilla(URL, force_rest=True, user=USER, api_key=API_KEY)
except (BugzillaError, RequestException) as exc:
    error = exc
    logging.error("%s", exc)
```

`ERROR:root:HTTPSConnectionPool(host='bugzilla.suse.com', port=443): Max retries exceeded with url: /rest/version?Bugzilla_api_key=MYAPIKEY (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x7f48458cbf90>: Failed to resolve 'bugzilla.suse.com' ([Errno -3] Temporary failure in name resolution)"))
`

The only workaround is to catch `requests.exceptions.ConnectionError`.

**Comment by ricardobranco777** [comment]
I have to use this workaround for now:

```
session = requests.Session()
api_key = creds.pop("api_key")
session.headers["Bugzilla_api_key"] = api_key
session.headers["X-Bugzilla-API-Key"] = api_key
client = Bugzilla(url=None, force_rest=True, requests_session=session, **creds)
try:
    client.connect(url)
except (BugzillaError, RequestException) as exc:
    logging.error("Bugzilla: %s: %s", url, exc)
```

**Comment by ricardobranco777** [comment]
Tracking in https://bugzilla.suse.com/show_bug.cgi?id=1215718

**Comment by crazyscientist** [comment]
Fixed by #188. Thank you :slightly_smiling_face:
